### ISSUE TITLE
Strange accidental horizontal placement case

### ISSUE THREAD

**Issue by craigsapp** [rism-ch/verovio#455]
Here is a strange case of extra horizontal offset given to an accidental, dependent on the height of the pitch above in the chord:

<img width="397" alt="screen shot 2017-01-24 at 20 05 33" src="https://cloud.githubusercontent.com/assets/3487289/22277485/ae862ac4-e270-11e6-9f14-020659024b85.png">

**Comment by craigsapp** [comment]
Probably a new bug, and it seems to be related to the layout system for accidentals thinking that `accid@accid.ges` is a displaying accidental:

<img width="240" alt="screen shot 2017-01-24 at 20 18 48" src="https://cloud.githubusercontent.com/assets/3487289/22277693/6fddc910-e272-11e6-8e03-18809fb20665.png">

MEI data:

```xml
<?xml version="1.0" encoding="UTF-8"?>
<?xml-model href="http://music-encoding.org/schema/3.0.0/mei-all.rng" type="application/xml" schematypens="http://relaxng.org/ns/structure/1.0"?>
<?xml-model href="http://music-encoding.org/schema/3.0.0/mei-all.rng" type="application/xml" schematypens="http://purl.oclc.org/dsdl/schematron"?>
<mei xmlns="http://www.music-encoding.org/ns/mei" meiversion="3.0.0">
 <meiHead>
  <fileDesc>
   <titleStmt>
    <title />
   </titleStmt>
   <pubStmt />
  </fileDesc>
  <encodingDesc>
   <appInfo>
    <application isodate="2017-01-24T20:22:05" version="0.9.14-dev-4fd8628-dirty">
     <name>Verovio</name>
     <p>Transcoded from Humdrum</p>
    </application>
   </appInfo>
  </encodingDesc>
  <workDesc>
   <work>
    <titleStmt>
     <title />
    </titleStmt>
   </work>
  </workDesc>
 </meiHead>
 <music>
  <body>
   <mdiv>
    <score>
     <scoreDef xml:id="scoredef-000000210354685">
      <staffGrp xml:id="m-000000031577701">
       <staffDef xml:id="staffdef-000000083216575" clef.shape="F" clef.line="4" n="1" lines="5" />
      </staffGrp>
     </scoreDef>
     <section xml:id="section-000000018976194">
      <measure xml:id="measure-L3" n="1">
       <staff xml:id="staff-L3F1" n="1">
        <layer xml:id="layer-L3F1" n="1">
         <chord xml:id="chord-L4F1" dur="4">
          <note xml:id="note-L4F1S1" oct="3" pname="c">
           <accid xml:id="accid-L4F1S1" accid="s" />
          </note>
          <note xml:id="note-L4F1S2" oct="3" pname="e">
           <accid xml:id="accid-L4F1S2" accid="s" />
          </note>
          <note xml:id="note-L4F1S3" oct="3" pname="g">
           <accid xml:id="accid-L4F1S3" accid="s" />
          </note>
         </chord>
         <chord xml:id="chord-L5F1" dur="4">
          <note xml:id="note-L5F1S1" oct="3" pname="c">
           <accid xml:id="accid-L5F1S1" accid="s" func="caution" />
          </note>
          <note xml:id="note-L5F1S2" oct="3" pname="e">
           <accid xml:id="accid-L5F1S2" accid="s" func="caution" />
          </note>
          <note xml:id="note-L5F1S3" oct="3" pname="g">
           <accid xml:id="accid-L5F1S3" accid.ges="s" />
          </note>
         </chord>
         <chord xml:id="chord-L6F1" dur="4">
          <note xml:id="note-L6F1S1" oct="3" pname="c">
           <accid xml:id="accid-L6F1S1" accid.ges="s" />
          </note>
          <note xml:id="note-L6F1S2" oct="3" pname="e">
           <accid xml:id="accid-L6F1S2" accid="s" func="caution" />
          </note>
          <note xml:id="note-L6F1S3" oct="3" pname="g">
           <accid xml:id="accid-L6F1S3" accid.ges="s" />
          </note>
         </chord>
        </layer>
       </staff>
      </measure>
     </section>
    </score>
   </mdiv>
  </body>
 </music>
</mei>
```


Removing the `accid` elements which contain `@accid.ges`:

<img width="443" alt="screen shot 2017-01-24 at 20 25 14" src="https://cloud.githubusercontent.com/assets/3487289/22277809/4df53c88-e273-11e6-8aa0-9866999af214.png">

MEI data:


```xml
<?xml version="1.0" encoding="UTF-8"?>
<?xml-model href="http://music-encoding.org/schema/3.0.0/mei-all.rng" type="application/xml" schematypens="http://relaxng.org/ns/structure/1.0"?>
<?xml-model href="http://music-encoding.org/schema/3.0.0/mei-all.rng" type="application/xml" schematypens="http://purl.oclc.org/dsdl/schematron"?>
<mei xmlns="http://www.music-encoding.org/ns/mei" meiversion="3.0.0">
 <meiHead>
  <fileDesc>
   <titleStmt>
    <title />
   </titleStmt>
   <pubStmt />
  </fileDesc>
  <encodingDesc>
   <appInfo>
    <application isodate="2017-01-24T20:22:05" version="0.9.14-dev-4fd8628-dirty">
     <name>Verovio</name>
     <p>Transcoded from Humdrum</p>
    </application>
   </appInfo>
  </encodingDesc>
  <workDesc>
   <work>
    <titleStmt>
     <title />
    </titleStmt>
   </work>
  </workDesc>
 </meiHead>
 <music>
  <body>
   <mdiv>
    <score>
     <scoreDef xml:id="scoredef-000000210354685">
      <staffGrp xml:id="m-000000031577701">
       <staffDef xml:id="staffdef-000000083216575" clef.shape="F" clef.line="4" n="1" lines="5" />
      </staffGrp>
     </scoreDef>
     <section xml:id="section-000000018976194">
      <measure xml:id="measure-L3" n="1">
       <staff xml:id="staff-L3F1" n="1">
        <layer xml:id="layer-L3F1" n="1">
         <chord xml:id="chord-L4F1" dur="4">
          <note xml:id="note-L4F1S1" oct="3" pname="c">
           <accid xml:id="accid-L4F1S1" accid="s" />
          </note>
          <note xml:id="note-L4F1S2" oct="3" pname="e">
           <accid xml:id="accid-L4F1S2" accid="s" />
          </note>
          <note xml:id="note-L4F1S3" oct="3" pname="g">
           <accid xml:id="accid-L4F1S3" accid="s" />
          </note>
         </chord>
         <chord xml:id="chord-L5F1" dur="4">
          <note xml:id="note-L5F1S1" oct="3" pname="c">
           <accid xml:id="accid-L5F1S1" accid="s" func="caution" />
          </note>
          <note xml:id="note-L5F1S2" oct="3" pname="e">
           <accid xml:id="accid-L5F1S2" accid="s" func="caution" />
          </note>
          <note xml:id="note-L5F1S3" oct="3" pname="g">
          </note>
         </chord>
         <chord xml:id="chord-L6F1" dur="4">
          <note xml:id="note-L6F1S1" oct="3" pname="c">
          </note>
          <note xml:id="note-L6F1S2" oct="3" pname="e">
           <accid xml:id="accid-L6F1S2" accid="s" func="caution" />
          </note>
          <note xml:id="note-L6F1S3" oct="3" pname="g">
          </note>
         </chord>
        </layer>
       </staff>
      </measure>
     </section>
    </score>
   </mdiv>
  </body>
 </music>
</mei>
```

**Comment by craigsapp** [comment]
Looks good, now the first MEI example renders like the second one:

<img width="416" alt="screen shot 2017-01-25 at 04 54 10" src="https://cloud.githubusercontent.com/assets/3487289/22291293/59e2449a-e2ba-11e6-8b35-a5359e41e1a5.png">

**Comment by ahwitz** [comment]
@lpugin : was this a problem with my code or with the change you'd made over the last few days?

**Comment by craigsapp** [comment]
I haven't used `note/accid@accid.ges` before, so it probably uncovered an existing bug in code that has never been used.

**Comment by lpugin** [comment]
@ahwitz nothing to do with the chords specifically. And indeed a bug that was already there

**Comment by ahwitz** [comment]
Roger. Glad it's fixed now!

**Comment by lpugin** [comment]
We still need to modify View::CalculateAccidX in order to take into account other types of accidentals and also enclose accidentals. I think we would need to rethink the accidSpace mechanism but you probably have a better view of this.

**Comment by ahwitz** [comment]
Enclosing I haven't thought about at all yet.

The work I did on accidentals during the last MEC is still available at https://github.com/ahwitz/verovio/commits/ahwitz-devel - I just haven't had the chance to do anything with it since then. I got a decent amount done, but then got stuck with something and I don't quite remember what. Can look back into that at some point soon if you want me to, but I don't know if it'd just be easier for you to salvage all the enums/giant switches I made and implement the logic the right way.
